docs/InstrumentationEvents.md

```diff
@@ -29,149 +29,40 @@ Instrumentation Event:
-    `size`,
-    `createdAt`,
-    `sentAt`,
-    `pendingDuration`,
-    `duration`,
-    `apiName`,
-    `apiKey`,
-    `apiVersion`
-  }
-
-* consumer.events.REQUEST_TIMEOUT
-  payload: {
-    `broker`,
-    `clientId`,
-    `correlationId`,
-    `createdAt`,
-    `sentAt`,
-    `pendingDuration`,
-    `apiName`,
-    `apiKey`,
-    `apiVersion`
-  }
-
-* consumer.events.REQUEST_QUEUE_SIZE
-  payload: {
-    `broker`,
-    `clientId`,
-    `queueSize`
-  }
+| event               | payload                                                                                                                                                                                                       | description |
+|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------|
+| REQUEST             | {`broker`, `clientId`, `correlationId`, `size`, `createdAt`, `sentAt`, `pendingDuration`, `duration`, `apiName`, `apiKey`, `apiVersion`} | Whenever a request is made to the broker. |
+| CONNECT             |                                                                                                                                                                                                               | Consumer connected with broker. |
+| GROUP_JOIN          | {`groupId`, `memberId`, `leaderId`, `isLeader`, `memberAssignment`, `groupProtocol`, `duration`}                                                                                                                   | Consumer has joined the group. |
+| FETCH_START         | {}                                                                                                                                                                                                            | Starting to fetch messages from broker. |
+| FETCH               | {`numberOfBatches`, `duration`}                                                                                                                                                                           | Fetched messages from the broker. |
+| START_BATCH_PROCESS | {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`}                                                                                | Starting to process the batch. |
+| END_BATCH_PROCESS   | {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`, `duration`}                                                                  | Processed the batch. |
+| COMMIT_OFFSETS      | {`groupId`, `memberId`, `groupGenerationId`, `topics`}                                                                                                                                                | Committed resolved offsets. |
```

Comment:
This is not necessarily true, because you can commit arbitrary offsets, not just resolved ones.

```suggestion
| COMMIT_OFFSETS      | {`groupId`, `memberId`, `groupGenerationId`, `topics`}                                                                                                                                                | Committed offsets. |
```